src/commands/dsyms/upload.ts

```diff
@@ -0,0 +1,172 @@
+import retry from 'async-retry'
+import chalk from 'chalk'
+import {Command} from 'clipanion'
+import {BufferedMetricsLogger} from 'datadog-metrics'
+import path from 'path'
+import asyncPool from 'tiny-async-pool'
+
+import {apiConstructor} from './api'
+import {InvalidConfigurationError} from './errors'
+import {APIHelper, Payload} from './interfaces'
+import {getMetricsLogger} from './metrics'
+import {
+  renderCommandInfo,
+  renderConfigurationError,
+  renderDryRunUpload,
+  renderFailedUpload,
+  renderRetriedUpload,
+  renderSuccessfulCommand,
+} from './renderer'
+import {getBaseIntakeUrl, getMatchingDSYMFiles, isZipFile, unzipToTmpDir} from './utils'
+
+import {ApiKeyValidator} from '../../helpers/apikey'
+import {UploadStatus} from '../../helpers/interfaces'
+
+const errorCodesNoRetry = [400, 403, 413]
+
+export class UploadCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Upload dSYM files to Datadog.',
+    details: `
+            This command will upload all dSYM files to Datadog in order to symbolicate crash reports received by Datadog.
+            See README for details.
+        `,
+    examples: [
+      ['Upload all dSYM files in Derived Data path', 'datadog-ci dsyms upload ~/Library/Developer/Xcode/DerivedData'],
+      [
+        'Upload all dSYM files in a zip file (this is usually the case if your app has Bitcode enabled)',
+        'datadog-ci dsyms upload /path/to/folder/my_file.zip',
+      ],
+    ],
+  })
+
+  private apiKeyValidator: ApiKeyValidator
+  private basePath?: string
```

Comment:
Is it really possible to not have it?
I see it marked as required.
```suggestion
  private basePath: string
```